### PR TITLE
Add HyperClockCache Java API.

### LINKED ISSUE #11510: HyperClockCache not supported in Java API
> Note: Please use Issues only for bug reports. For questions, discussions, feature requests, etc. post to dev group: https://groups.google.com/forum/#!forum/rocksdb or https://www.facebook.com/groups/rocksdb.dev

### Expected behavior

I have upgraded rocksdb to 8.1.1 primarily to use ClockCache instead of LRUCache since it's now production ready.

### Actual behavior

However, upon investigation I found out that even when I use `ClockCache` in java, rocksDB still creates an `LRUCache`

The reason for this seems to be some API incompatibility between HyperClockCache and existing LRUCache

Can you please implement this since I have seen the benchmarks and it will give atlease 1.5-2X performance boost for us.

Screenshot 2023-06-05 at 6 24 08 PM

### Steps to reproduce the behavior

Just use `new ClockCache()` in java program

### PR DESCRIPTION
**Pull request by rhubner**
Fix #11510

### CODE REVIEW

File: java/src/test/java/org/rocksdb/HyperClockCacheTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  public void newHyperClockCache() throws RocksDBException {
+    RocksDB.loadLibrary();
+    try (Cache cache = new HyperClockCache(1024 * 1024, 0, 8, false)) {
+      BlockBasedTableConfig tableConfing = new BlockBasedTableConfig();
+      tableConfing.setBlockCache(cache);
+      try (Options options = new Options()) {
+        options.setTableFormatConfig(tableConfing);
+        options.setCreateIfMissing(true);
+        try (RocksDB db = RocksDB.open(options, dbFolder.getRoot().getAbsolutePath())) {
+          db.put("testKey".getBytes(), "testData".getBytes());
```
Comment (alanpaxton) [inline java/src/test/java/org/rocksdb/HyperClockCacheTest.java:20]: Would it be a good idea to call the `cache.getUsage()` and `cache.getPinnedUsage()` methods of the `Cache` superclass, as the `LRUCacheTest` does ? Just to sanity check the calls and give some confidence that the cache object is valid

**Review by alanpaxton — COMMENTED** [review]
LGTM - I've suggested doing a bit more in the test, to sanity check the cache object we are holding

File: java/src/test/java/org/rocksdb/HyperClockCacheTest.java (hunk 1 above)
Comment (rhubner) [reply java/src/test/java/org/rocksdb/HyperClockCacheTest.java:20]: Good point. It looks like it works. Do you think we can put it inside assert? 
```java
assertThat(cache.getUsage()).isNotZero();
assertThat(cache.getPinnedUsage()).isNotZero();
```

Comment (alanpaxton) [reply java/src/test/java/org/rocksdb/HyperClockCacheTest.java:20]: Yes. See LRUCacheTest.java

**Review by cbi42 — APPROVED** [review]

**Pull request closed without merging**